### PR TITLE
Build Home Page - W50-220

### PR DESCRIPTION
**Pull request by ridham-virufy**
Developed the “Homepage” web page based on the provided Figma design.

[Figma Mobile version](https://www.figma.com/design/CJVubKy4QZETjVty3Ak4eK/Website-6.0?node-id=671-5008&t=JkiBqk6C297k46Ly-1)
[Figma Desktop Version](https://www.figma.com/design/CJVubKy4QZETjVty3Ak4eK/Website-6.0?node-id=331-1815&t=wMetGCxZY361l2E6-1)

Mobile Screenshots for reference:
Screenshot 2024-06-11 at 10 22 10 AM
Screenshot 2024-06-11 at 10 22 30 AM
Screenshot 2024-06-11 at 10 22 36 AM

Let me know if there are any suggestions!

I do need help figuring out why the header and footer are not looking the desired way, that is having a transparent background. It might work once Homepage and the updated header by @a-wong-8 is merged together into the release.

### CODE REVIEW

**Review by a-wong-8 — COMMENTED** [review]
good job @ridham-virufy ! visually, the only problem I see so far is the text running into the img in tablet view. I havent looked at the code yet. 

Screenshot 2024-06-11 at 9 52 31 AM

**Comment by gevuong** [comment]
can a max width be set for the page? That way, the content will not continue to stretch beyond a certain width. I see this occurring at 2560px (4k). I suggest using a max width of `max-w-5xl`.

For context:
Screenshot 2024-06-13 at 5 55 50 PM
Screenshot 2024-06-13 at 5 57 46 PM

Other pages:
Screenshot 2024-06-13 at 5 57 22 PM
Screenshot 2024-06-13 at 5 56 37 PM

**Review by a-wong-8 — COMMENTED** [review]
the code looks good to me. only thing is fix the thing I mentioned above 👍🏼

File: src/components/organisms/section-10/Section10.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import ImageSection3 from '~/assets/static/images/section3/Algorithm.jpg';
+import ImageAtomLocal from '~/components/atoms/imageAtom/ImageAtomLocal';
+
+
+export default function Section2({
+    title,
+    subtitle,
+    text
+}) {
+  const { } = useI18n();
```
Comment (gevuong) [inline src/components/organisms/section-10/Section10.js]: Remove if unused, including its import

File: src/pages/home.tsx [hunk 2]
```diff
@@ -1,106 +1,57 @@
...
-          ImageStyleProps=""
-          //Buttons
-          buttons={buttons}
-          buttonTechnology={false}
+      <div className="w-full overflow-hidden">
+        <Section2
+          sizeImg="width-100vw"
+          borderImg="none"
+          src={ImageSection2}
+          altImg="Imagen cabezera"
```
Comment (gevuong) [inline src/pages/home.tsx]: should this be in English?

File: src/components/organisms/section-3/Section3.js [hunk 3]
```diff
@@ -1,51 +1,97 @@
...
+    title1,
+    sub1,
+    title2,
+    sub2,
+    title3,
+    sub3,
+    disclaimer,
+    buttonText
+}) {
+  const { } = useI18n();
```
Comment (gevuong) [inline src/components/organisms/section-3/Section3.js]: remove if unused, including its import

**Comment by gevuong** [comment]
Looks good! Consider setting a max width to prevent content from stretching, and to match the other pages. Note that the hardcoded content will eventually need to be abstracted to support other languages (similar to the other updated pages). But that can be done in a future revision.

**Comment by ridham-virufy** [comment]
- Made the content aligned centered and set a max width.

image


- Now the text is not running into the img in tablet view

Screenshot 2024-06-18 at 11 39 13 AM

- Removed extra lines of code related to the internationalization of the website using different languages.

- Changed the alt text for the homepage image in section 2 from Spanish to English. 

**Once the merge is approved it can be moved to QA**

**Review by a-wong-8 — COMMENTED** [review]
looks like there is a new problem now after 1440px:
Screenshot 2024-06-18 at 6 07 07 PM

**Comment by ridham-virufy** [comment]
> looks like there is a new problem now after 1440px: Screenshot 2024-06-18 at 6 07 07 PM

It is fixed now, Should I move it to QA now?

**Comment by gevuong** [comment]
So at 4k, the "Your Health, Our Priority" blue card stretches. If it's possible and you have time, I would set a max width for it. For example, if you look at `/our-story`, the card in the Values section does not stretch beyond a max width: https://deploy-preview-30--creative-jalebi-f8e18b.netlify.app/our-story

Screenshot 2024-06-19 at 1 46 13 PM

It looks like the "How It Works" section stretches as well. If it's possible and you have time, I would set a max width for that section so that it doesn't stretch to the edge of the screen for larger screens (ie. 4k), and it will more so align with the Figma design and other sections (ie. hero, algorithms).
Screenshot 2024-06-19 at 1 46 00 PM

Other than that, great job.

**Comment by ridham-virufy** [comment]
> So at 4k, the "Your Health, Our Priority" blue card stretches. If it's possible and you have time, I would set a max width for it. For example, if you look at `/our-story`, the card in the Values section does not stretch beyond a max width: https://deploy-preview-30--creative-jalebi-f8e18b.netlify.app/our-story
> 
> Screenshot 2024-06-19 at 1 46 13 PM
> It looks like the "How It Works" section stretches as well. If it's possible and you have time, I would set a max width for that section so that it doesn't stretch to the edge of the screen for larger screens (ie. 4k), and it will more so align with the Figma design and other sections (ie. hero, algorithms). Screenshot 2024-06-19 at 1 46 00 PM
> 
> Other than that, great job.

I have fixed the excess stretching of "How it works", and it should be smooth and good now. 

For "Your Health, Our Priority" card, making it not stretch was my implementation at first but later I changed it as it was not looking good to me on a personal level. If you observe the Our Values card in the deploy link you gave, it has a good background image for the emptiness and hence, it makes sense to not make it stretch. Here the background is black color and hence, it is not appealing enough. (Refer the image below)
image

We can do two things now:
1) Ask UI team to update the design to have a background image for that section
or
2) decide if we want the blue card to stretch or not stretch with a black background.

**Comment by gevuong** [comment]
I see. Sure lets just leave it as-is for now with the blue card stretching, and we can ask the UI team if they want the blue card to stretch or not during QA.

**Comment by gevuong** [comment]
@a-wong-8 and @ridham-virufy, should the responsive webpages we build be supporting Mobile S - 320px as well?
Currently, I believe all the webpages Anthony and I have built support screens from 320px to 4k, so for consistency, I think this page should do so as well. What do you all think? I'm asking because I notice there is overlapping text at Mobile S - 320px.

Here's link to the Slack discussion regarding 320px. We have since fixed those issues mentioned in the thread: https://virufy.slack.com/archives/C076DU42PRQ/p1715888952371919

**Comment by a-wong-8** [comment]
@gevuong I say yes to 320px

**Comment by ridham-virufy** [comment]
> @a-wong-8 and @ridham-virufy, should the responsive webpages we build be supporting Mobile S - 320px as well? Currently, I believe all the webpages Anthony and I have built support screens from 320px to 4k, so for consistency, I think this page should do so as well. What do you all think? I'm asking because I notice there is overlapping text at Mobile S - 320px.
> 
> Here's link to the Slack discussion regarding 320px. We have since fixed those issues mentioned in the thread: https://virufy.slack.com/archives/C076DU42PRQ/p1715888952371919

Made edits so that it works in 320px viewport! 

Screenshot 2024-06-19 at 6 46 39 PM
Screenshot 2024-06-19 at 6 49 33 PM
image

**Comment by ridham-virufy** [comment]
> > So at 4k, the "Your Health, Our Priority" blue card stretches. If it's possible and you have time, I would set a max width for it. For example, if you look at `/our-story`, the card in the Values section does not stretch beyond a max width: https://deploy-preview-30--creative-jalebi-f8e18b.netlify.app/our-story
> > Screenshot 2024-06-19 at 1 46 13 PM
> > It looks like the "How It Works" section stretches as well. If it's possible and you have time, I would set a max width for that section so that it doesn't stretch to the edge of the screen for larger screens (ie. 4k), and it will more so align with the Figma design and other sections (ie. hero, algorithms). Screenshot 2024-06-19 at 1 46 00 PM
> > Other than that, great job.
> 
> I have fixed the excess stretching of "How it works", and it should be smooth and good now.
> 
> For "Your Health, Our Priority" card, making it not stretch was my implementation at first but later I changed it as it was not looking good to me on a personal level. If you observe the Our Values card in the deploy link you gave, it has a good background image for the emptiness and hence, it makes sense to not make it stretch. Here the background is black color and hence, it is not appealing enough. (Refer the image below) image
> 
> We can do two things now:
> 
> 1. Ask UI team to update the design to have a background image for that section
>    or
> 2. decide if we want the blue card to stretch or not stretch with a black background.

Sherrie suggested keeping the "Your Health, Our Priority" card to stretch, so the code is updated to do so.

**Comment by gevuong** [comment]
Can you remove the `/home` route? Clicking `Home` on the navbar should redirect to `/`. It doesn't need to be appended with `/home` because the content at `/` is the same.

Update: Has been resolved in a separate merged PR.

**Comment by gevuong** [comment]
>I do need help figuring out why the header and footer are not looking the desired way, that is having a transparent background. It might work once Homepage and the updated header by @a-wong-8 is merged together into the release.

We should try to figure this out before merging the branch. Can you elaborate on what you mean by the footer is not looking the desired way?

**Comment by a-wong-8** [comment]
@ridham-virufy it looks like you don't have the latest NAV bar and footer. Try running git pull origin <branch name>

**Comment by ridham-virufy** [comment]
> > I do need help figuring out why the header and footer are not looking the desired way, that is having a transparent background. It might work once Homepage and the updated header by @a-wong-8 is merged together into the release.
> 
> We should try to figure this out before merging the branch. Can you elaborate on what you mean by the footer is not looking the desired way?

I pulled all the commits, but I don't think it helped in updation of the navbar and footer.
image

**Comment by gevuong** [comment]
@ridham-virufy how are you pulling changes from release? Those changes you pulled are appearing as new changes in this PR and adding 20 commits..

After pulling, the commit should look more like this, as shown in this PR https://github.com/virufy6/virufy6.github.io/pull/36:
Screenshot 2024-06-26 at 9 31 41 AM

Did you run `git pull origin release`?

_Edit: I'm also noticing this in your other PR as well. It's adding a lot of unnecessary code changes and 30 commits, making PR https://github.com/virufy6/virufy6.github.io/pull/34 more difficult to review and determine what were your code changes and what wasn't. If possible, I would either_

_1. revert how you git pulled in both of your branches_
_2. or create a new branch with only your changes, then run `git pull origin release`._

**Comment by ridham-virufy** [comment]
> @ridham-virufy how are you pulling changes from release? Those changes you pulled are appearing as new changes in this PR and adding 20 commits..
> 
> After pulling, the commit should look more like this, as shown in this PR #36: Screenshot 2024-06-26 at 9 31 41 AM
> 
> Did you run `git pull origin release`?
> 
> _Edit: I'm also noticing this in your other PR as well. It's adding a lot of unnecessary code changes and commits, making PR #34 more difficult to review and determine what were your code changes and what wasn't. If possible, I would either_ _1. revert how you git pulled in both of your branches_ _2. or create a new branch with only your changes, then run `git pull origin release`._

Yes, the command I ran was `git pull origin release` I am not sure why all of those changes are reflected in the commit history. I can revert both the pull's but is there another way you want me to pull recent push from release? 
Do you think trying to merge the header footer branch into mine and then pushing it to release will fix the deploy preview's header and footer?

**Comment by gevuong** [comment]
@ridham-virufy
1. Can you first revert the git pull so we can have your branch in the state it originally was?
2. Since you pulled from release, it should already contain the header and footer changes. If the issue still appears in preview, then that did not fix the transparent issue.
3. I asked this earlier but haven't received a response: Can you elaborate on what you mean by the footer is not looking the desired way?

**Comment by ridham-virufy** [comment]
> @ridham-virufy
> 
> 1. Can you first revert the git pull so we can have your branch in the state it originally was?
> 2. Since you pulled from release, it should already contain the header and footer changes. If the issue still appears in preview, then that did not fix the transparent issue.
> 3. I asked this earlier but haven't received a response: Can you elaborate on what you mean by the footer is not looking the desired way?

1. I have reverted all the extra commits.
2. I started working on Home page before Anthony was done with Header so, when I started the header was not completed. 
3. Footer is perfect, the only issues is transparency of the Header.

**Comment by a-wong-8** [comment]
@ridham-virufy this is how you fix the navbar issue in home.tsx:
```
    <div className="flex flex-col items-center justify-center **-mb-24**">
      <div className="**relative -top-24** w-full overflow-hidden">

```
Screenshot 2024-06-26 at 12 57 17 PM

**Comment by gevuong** [comment]
@ridham-virufy Thank you for reverting the commits. It helps me debug the issue. Ok I think what's going on is the branches you create are not branched off of `release`, which may explain why every time you pull from `release`, those changes you pull are considered new changes..

When you create a new branch, it needs to branch off of `release`. One way to do that is to first make sure you're on the `release` branch, run `git checkout -b <insert new branch name>`. That newly created branch should then be a branch of `release`.

_Edit: From there, I would add your page changes to that branch. I would do the same with the other PR._

**Comment by ridham-virufy** [comment]
> @ridham-virufy Thank you for reverting the commits. It helps me debug the issue. Ok I think what's going on is the branches you create are not branched off of `release`, which may explain why every time you pull from `release`, those changes you pull are considered new changes..
> 
> When you create a new branch, it needs to branch off of `release`. One way to do that is to first make sure you're on the `release` branch, run `git checkout -b <insert new branch name>`. That newly created branch should then be a branch of `release`.
> 
> _Edit: From there, I would add your page changes to that branch. I would do the same with the other PR._



> @ridham-virufy Thank you for reverting the commits. It helps me debug the issue. Ok I think what's going on is the branches you create are not branched off of `release`, which may explain why every time you pull from `release`, those changes you pull are considered new changes..
> 
> When you create a new branch, it needs to branch off of `release`. One way to do that is to first make sure you're on the `release` branch, run `git checkout -b <insert new branch name>`. That newly created branch should then be a branch of `release`.
> 
> _Edit: From there, I would add your page changes to that branch. I would do the same with the other PR._

I have been creating branches manually as shown in the screenshot below. I am pretty sure it is branched off from release. 
Screenshot 2024-06-26 at 3 43 47 PM

Let me know what you want me to do.

**Comment by gevuong** [comment]
Ok here, I just ran `git pull origin release` on your branch and it worked..
Screenshot 2024-06-26 at 3 08 48 PM

I think it may have to do with the merge strategy you have set up. To confirm, can you share screenshot of the output when you run `git config --list`?

**Comment by ridham-virufy** [comment]
> git config --list

image
Does it look good?

**Comment by gevuong** [comment]
Ok I figured it out. You are rebasing on a public branch (ie. `release`) which goes agains the rule of rebasing. What that will do is all the commits from `release` will be added on top of your feature branch. If you're not familiar with rebasing and when to use it, I recommend setting your pull rebase config to false. To do this, run `git config pull.rebase false`.

Now, whenever you pull from a public branch (ie. `release`), all the commits from `release` will not be added on top of your feature branch. Instead, it will merge and occupy a single commit.

**Review by gevuong — APPROVED** [review]

**Pull request merged**